core/deployment/src/main/java/io/quarkus/deployment/pkg/steps/NativeImageBuildRemoteContainerRunner.java

```diff
@@ -34,59 +39,93 @@ protected void preBuild(Path outputDir, List<String> buildArgs) throws Interrupt
         final List<String> containerRuntimeArgs = Arrays.asList("-v",
                 CONTAINER_BUILD_VOLUME_NAME + ":" + NativeImageBuildStep.CONTAINER_BUILD_VOLUME_PATH);
         final String[] createTempContainerCommand = buildCommand("create", containerRuntimeArgs, Collections.emptyList());
-        containerId = runCommandAndReadOutput(createTempContainerCommand, "Failed to create temp container.");
+        try {
+            containerId = runCommandAndReadOutput(createTempContainerCommand).get(0);
+        } catch (RuntimeException | InterruptedException | IOException e) {
+            throw new RuntimeException("Failed to create temp container.", e);
+        }
         // docker cp <files> <containerID>:/project
-        String[] copyCommand = new String[] { containerRuntime.getExecutableName(), "cp", outputDir.toAbsolutePath() + "/.",
+        final String[] copyCommand = new String[] {
+                containerRuntime.getExecutableName(), "cp", outputDir.toAbsolutePath() + "/.",
                 containerId + ":" + NativeImageBuildStep.CONTAINER_BUILD_VOLUME_PATH };
-        runCommand(copyCommand, "Failed to copy source-jar and libs from host to builder container", null);
+        runCommand(copyCommand, "Failed to copy source-jar and libs from host to builder container");
         super.preBuild(outputDir, buildArgs);
     }
 
-    private String runCommandAndReadOutput(String[] command, String errorMsg) throws IOException, InterruptedException {
+    private List<String> runCommandAndReadOutput(String[] command) throws IOException, InterruptedException {
         log.info(String.join(" ", command).replace("$", "\\$"));
-        Process process = new ProcessBuilder(command).start();
+        final Process process = new ProcessBuilder(command).start();
         if (process.waitFor() != 0) {
-            throw new RuntimeException(errorMsg);
+            throw new RuntimeException("Command failed: " + String.join(" ", command));
         }
         try (BufferedReader reader = new BufferedReader(new InputStreamReader(process.getInputStream()))) {
-            return reader.readLine();
+            return reader.lines().toList();
```

Comment:
What's the motivation behind this change? I understand that it's more generic this way but since we only use the method in a single place where this is not needed it seems to unnecessarily (for the time being) complicate things.

Reply:
@zakkak I used that to read the output of `exec` for `find` to list the .so files in the started container. Then I abandoned that approach in favor of using the `-H:+GenerateBuildArtifactsFile`. So strictly speaking, we can leave in the thing that returns only the first line of output, however confusing that might be.